### ISSUE TITLE
Insecure private repositories fail installation

### ISSUE THREAD

**Issue by seansfkelley** [sdispater/poetry#503]
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: macOS 10.14
- **Poetry version**: 0.12.0

## Issue

Trying to install a dependency from an HTTP (rather than HTTPS) private repo causes `install` to fail complaining about missing authentication-related parameters. I have not specified because there is no authentication necessary. This was not an issue in 0.11.x, which did not demand authentication.

```toml
[tool.poetry]
name = "foo"
version = "0.1.0"
description = ""
authors = ["foo <foo@foo.com>"]
license = ""

[[tool.poetry.source]]
name = "pypicloud"
url = "http://<hostname>/simple/"

[tool.poetry.dependencies]
python = "^3.6"
my-private-package = "^x.y.z"
```

with `-vvv`:

```
> poetry install -vvv 
Using virtualenv: /Users/seankelley/Library/Caches/pypoetry/virtualenvs/foo-py3.6
Installing dependencies from lock file


Package operations: 7 installs, 0 updates, 0 removals

  - Installing markupsafe (1.0)
    Installing from unsecure host: <hostname>                                                                                   
[TypeError]                                                         
get_http_basic_auth() missing 1 required positional argument: 'repository_name'  
                                                                                   
Exception trace:
 /Users/seankelley/.poetry/lib/poetry/_vendor/py3.6/cleo/application.py in run() at line 94
   status_code = self.do_run(input_, output_)
 /Users/seankelley/.poetry/lib/poetry/console/application.py in do_run() at line 88
   return super(Application, self).do_run(i, o)
 /Users/seankelley/.poetry/lib/poetry/_vendor/py3.6/cleo/application.py in do_run() at line 197
   status_code = command.run(input_, output_)
 /Users/seankelley/.poetry/lib/poetry/console/commands/command.py in run() at line 77
   return super(BaseCommand, self).run(i, o)
 /Users/seankelley/.poetry/lib/poetry/_vendor/py3.6/cleo/commands/base_command.py in run() at line 146
   status_code = self.execute(input_, output_)
 /Users/seankelley/.poetry/lib/poetry/_vendor/py3.6/cleo/commands/command.py in execute() at line 107
   return self.handle()
 /Users/seankelley/.poetry/lib/poetry/console/commands/install.py in handle() at line 56
   return_code = installer.run()
 /Users/seankelley/.poetry/lib/poetry/installation/installer.py in run() at line 76
   self._do_install(local_repo)
 /Users/seankelley/.poetry/lib/poetry/installation/installer.py in _do_install() at line 286
   self._execute(op)
 /Users/seankelley/.poetry/lib/poetry/installation/installer.py in _execute() at line 294
   getattr(self, "_execute_{}".format(method))(operation)
 /Users/seankelley/.poetry/lib/poetry/installation/installer.py in _execute_install() at line 319
   self._installer.install(operation.package)
 /Users/seankelley/.poetry/lib/poetry/installation/pip_installer.py in install() at line 48
   auth = get_http_basic_auth(package.source_reference)

install [--no-dev] [--dry-run] [-E|--extras EXTRAS] [--develop DEVELOP]
```

I tried to use a workaround by doing something like putting in fake username/passwords to see if that would bypass the problematic code, but the docs on this at https://poetry.eustace.io/docs/repositories/ are a little thin and it's not clear to me how to configure my `pyproject.toml` to use authentication for a private repo. The docs say:

> If your private repository requires HTTP Basic Auth be sure to add the username and password to your http-basic config using the example above

I can't find where the `poetry config` command listed in those docs makes changes, so I can't even try to mimic those changes in `pyproject.toml`.

**Comment by sdispater** [comment]
Sorry about that!

I know where the issue is coming from and I will have a fix shortly.

**Comment by sdispater** [comment]
This should be fixed in the new release (`0.12.2`)

**Comment by seansfkelley** [comment]
Yup, appears to work, thanks again for the quick fix!
